packages/form-js-editor/src/features/properties-panel/entries/ConditionEntry.js

```diff
@@ -0,0 +1,95 @@
+import { useRef } from 'preact/hooks';
+import { FeelEntry, isFeelEntryEdited } from '@bpmn-io/properties-panel';
+import { get } from 'min-dash';
+
+import { getSchemaVariables } from '@bpmn-io/form-js-viewer';
+
+import { useService } from '../hooks';
+import { usePrevious } from '../../../render/hooks';
+
+
+
+export function ConditionEntry(props) {
+  const {
+    editField,
+    field
+  } = props;
+
+  return [
+    {
+      id: 'condition',
+      component: Condition,
+      editField: editField,
+      field: field,
+      isEdited: isFeelEntryEdited
+    }
+  ];
+}
+
+
+function Condition(props) {
+  const {
+    editField,
+    field,
+    id
+  } = props;
+
+  const debounce = useService('debounce');
+
+  const variables = useVariables();
+
+  const path = [ 'condition' ];
+
+  const getValue = () => {
+    return get(field, path, '');
+  };
+
+  const setValue = (value) => {
+    return editField(field, { condition: value });
+  };
+
+  return FeelEntry({
+    debounce,
+    description: 'Condition under which the field is displayed',
+    element: field,
+    feel: 'required',
+    getValue,
+    id,
+    label: 'Show if',
+    setValue,
+    variables
+  });
+}
+
+// workaround to keep stable reference to variables array
+function useVariables() {
+  const form = useService('formEditor');
+  const schema = form.getSchema();
+
+  const variablesRef = useRef([]);
+
+  const variables = getSchemaVariables(schema);
+  const previousVariables = usePrevious(variables) || [];
+
+  const variablesChanged = !areEqual(variables, previousVariables);
+
+  if (variablesChanged) {
+    variablesRef.current = variables.map(key => ({ name: key }));
+  }
+
+  return variablesRef.current;
+}
```

Comment:
Should we move this under `/hooks`? I can imagine we will need this more often in the future (similar to `useCondition` in the `viewer`)

Reply:
Hmm I think I could simplify this a bit...

Reply:
OK it's now simplified and exported from hooks.

Reply:
I think we have to be careful here. The `useService` hook that is used in the new `useVariables` hook now needs a `FormEditorContext` to work correctly. Note that the properties panel can also be rendered outside of the editor. We use our own `useService` therefore: https://github.com/bpmn-io/form-js/blob/develop/packages/form-js-editor/src/features/properties-panel/hooks/usePropertiesPanelService.js

We either need the `formEditor` to be provided as a parameter for this hook or, alternatively, move the `useVariables` to the properties panels hooks.

Reply:
Yeah this is probably why the tests started to fail after the change.

Reply:
OK it should be fine now.